x-pack/plugins/security/kibana.json

```diff
@@ -3,7 +3,7 @@
   "version": "8.0.0",
   "kibanaVersion": "kibana",
   "configPath": ["xpack", "security"],
-  "requiredPlugins": ["features", "licensing"],
+  "requiredPlugins": ["data", "features", "home", "licensing", "management"],
```

Comment:
__note:__ I assume `data` and `management` are required plugins, but I'm curious if `home` can be optional - in legacy it was kind of required so I kept it as required here as well.

Reply:
I think `home` can be listed as an optional dependency. We're treating it that way for the Spaces plugin, so it would at least be consistent between our two main plugins.

I'm torn on the `data` and `management` plugins... currently, `data` is only used to retrieve index patterns, and index patterns are only used to offer helpful defaults in the "index privilege" dropdown of the role management screen. We can certainly work without those entries present, but I don't think it's necessarily worthwhile to introduce the added complexity of supporting this scenario. Disabling `data` will likely cause most of Kibana itself to become unusable in the future...

For `management` -- a lot of what the Security plugin has to offer is done via its Management screens, but I could see certain customers/users wanting to disable management, but still take advantage of our security features. They could, for example, manage their users and roles via our API, while still allowing users to authenticate and use Kibana normally. I think our ideal answer would be to use Feature Controls to disable management features, but that's not a reality for us yet (https://github.com/elastic/kibana/issues/35965)

Reply:
> I think home can be listed as an optional dependency. We're treating it that way for the Spaces plugin, so it would at least be consistent between our two main plugins.

++, it makes sense, will do (even though Kibana can't start without `home` right now since legacy Kibana plugin assumes that `home` is always present :see_no_evil: , I don't think it's correct @TinaHeiligers?).

> Disabling data will likely cause most of Kibana itself to become unusable in the future...

Yeah, and looking at its name it feels like it's a some sort of a fundamental plugin and I can't think of reason why user would want to disable it.

> For management -- a lot of what the Security plugin has to offer is done via its Management screens, but I could see certain customers/users wanting to disable management, but still take advantage of our security features. They could, for example, manage their users and roles via our API, while still allowing users to authenticate and use Kibana normally. I think our ideal answer would be to use Feature Controls to disable management features, but that's not a reality for us yet (#35965)

That's an interesting thought, and `management` is clear "domain" that can potentially be disabled. It costs nothing for us to make it optional - let's try (same story here though - Kibana just fails if `management` is disabled - that's bad and we should change that `all or nothing` behaviour and enforce during review).